[CBRD-26060] la_apply_xxx_log 함수 리팩토링

youngjun9072 opened the pull request:
<http://jira.cubrid.org/browse/CBRD-26060>

la_apply_xxx_log() 함수를 리팩토링합니다

### Description

1. 다음 함수들에 대해 리팩토링을 진행합니다.
la_apply_insert_log()
la_apply_update_log()
la_apply_delete_log()
2. 이 함수들은 오퍼레이션에 맞게 item, recdes를 가져와 오브젝트를 생성하고 리플로그에 메다는 역할을 수행합니다.
3. 이 과정에서 la_enable_sql_logging 가 설정되어있다면 파일로 쓰는 역할도 동시에 수행합니다.
4. 함수를 읽는 과정에서 파일 로깅부분의 코드가 난잡해 별도의 함수로 분리하려합니다
5. 또한 delete문을 기록하는 루틴에선 null 체크를 하지 않는 코드가 발견되어 보완합니다.

### Remarks

N/A

youngjun9072 committed:
la_apply_insert_log(), la_apply_update_log(), la_apply_delete_log() Function refactoring

- Added la_write_update_sql_log, la_write_insert_sql_log, and la_write_delete_sql_log functions and refactored the code that writes SQL statements to a file under the control of the la_enable_sql_logging flag.
- Extracted error‐handling logic into its own function.
- Added a null‐pointer check for mclass in the delete operation.

hornetmj commented on src/transaction/log_applier.c:
```diff
@@ -4963,53 +5009,82 @@ la_apply_delete_log (LA_ITEM * item)
     {
       assert (er_errid () != NO_ERROR);
-      error = er_errid ();
+      return er_errid ();
```
기존 동작과 같으려면, return 전에 la_log_apply_error()를 호출해 주어야 합니다. 또는, 에러 처리를 위해 goto error; 분기를 만들고, 모든 에러 처리는 error: 레이블에서 수행하는 것도 좋을 것 같습니다.

```suggestion
      la_log_apply_error ("apply_delete", ER_HA_LA_FAILED_TO_APPLY_DELETE, item, error);
      return er_errid ();
```

youngjun9072 replied:
@hornetmj 리뷰 반영했습니다. 다른 함수와 마찬가지로 goto end; 추가했습니다.

hornetmj commented on src/transaction/log_applier.c:
```diff
@@ -4963,53 +5009,82 @@ la_apply_delete_log (LA_ITEM * item)
-	    }
-	}
+static int
+la_write_update_sql_log (LA_ITEM * item, DB_OBJECT * class_obj, RECDES * recdes)
```
NIT:

같은 목적성 함수인 la_write_delete_sql_log(), la_write_update_sql_log(), la_write_insert_sql_log()는 함께 배치하는게 좋을 것 같습니다. 추가적으로, la_log_sql_error() 역시도 sql log와 관련된 함수이기 때문에 네이밍 규칙(ex) la_set_error_sql_log() 등)이 통일되면 더 가독성이 좋을 것 같습니다.

youngjun9072 replied:
@hornetmj la_log_sql_error -> la_set_error_sql_log 변경 반영했습니다.
함수들의 위치는 호출하는 함수의 바로 위에 있어야 나중에 코드를 읽고 찾을때 편할것같아 해당 위치에 배치했는데 다른 리뷰어의 추가 의견있으면 취합해보겠습니다.

hyahong replied:
개인적으로 코드 자체를 읽을 때는 바로 위에 있는 게 편한 것 같습니다. 다만, 그러려면 비슷한 성질/네이밍을 가진 함수의 선언은 모여있어야 따라가기 편할 것 같습니다.

hornetmj replied:
@youngjun9072 넵. 저의 경우는 c 구현의 경우 추후 sql_log와 관련된 작업을 확장할 때 관련 함수가 함께 모여있지 않으면 확인하기 번거롭거나 해당 함수가 존재하는지 몰라 유사한 함수를 구현하게 되는 경험들이 있습니다. 참고하셔서 정리해 주시면 될 것 같습니다. @hyahong님 의견도 좋은 포인트 같습니다.

youngjun9072 committed:
Apply review feedback

- Modified error handling in la_apply_delete_log function (added goto end:)
- Renamed function la_log_sql_error() to la_set_error_sql_log()

hyahong commented on src/transaction/log_applier.c:
```diff
@@ -5231,6 +5237,60 @@ la_is_mvcc_class (const OID * class_oid)
 }
 
+static bool
+la_write_insert_sql_log (LA_ITEM * item, DB_OBJECT * class_obj, RECDES * recdes)
```
반환형은 bool이지만, 내부에서는 int로 정의된 에러 코드를 반환하고 있습니다.

hyahong replied:
la_write_insert_sql_log와 la_write_update_sql_log의 prototype이 코드 상단에 추가되어야 할 것 같습니다.

youngjun9072 committed:
la_write_insert_sql_log function from bool to int

InChiJun commented on src/transaction/log_applier.c:
```diff
@@ -4935,22 +4935,68 @@ la_repl_add_object (MOP classop, LA_ITEM * item, RECDES * recdes)
+  er_log_debug (ARG_FILE_LINE,
+		"%s : error %d %s\n\tclass %s key %s\n", op_name, error, er_msg (), item->class_name, sb.get_buffer ());
+#endif
+
```
4965번 라인은 삭제해도 되지 않을까요?

youngjun9072 replied:
구두로 설명드렸듯이 #if 부터 #endif 까지를 하나의 문맥으로 인지하고 작업하였습니다.

youngjun9072 committed:
Add function prototype

- la_log_apply_error ()
- la_set_error_sql_log ()
- la_write_delete_sql_log ()
- la_write_update_sql_log()
- la_write_insert_sql_log()

youngjun9072 committed:
lint

youngjun9072 commented:
/run sql medium

youngjun9072 commented:
/run sql medium
